src/main/scala/net/snowflake/spark/snowflake/LoggerWrapper.scala

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright 2015-2020 Snowflake Computing
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.snowflake.spark.snowflake
+
+import org.slf4j.helpers.MessageFormatter
+import org.slf4j.{Logger, LoggerFactory}
+
+/**
+  * Logger Wrapper class.
+  *
+  * The logger wrapper may do some special handling for the log message,
+  * for example, sending telemetry messages. And then pass through the logging
+  * request to internal logger object.
+  *
+  * NOTE: LoggerWrapper only implements the functions which are used by Spark
+  *       Connector. If other functions in <code>org.slf4j.Logger</code> need
+  *       to be used, they need to be implemented in this class.
+  */
+private[snowflake] class LoggerWrapper(logger: Logger) {
+
+  // Do NOT throw exception when sending log entry
+  private def sendLogTelemetryIfEnabled(level: String, msg: String): Unit = {
+    try {
+      // Send the message as telemetry
```

Comment:
To reviewers:
1. The LoggerWrapper will get a TelemetryReporter instance dynamically for telemetry message, instead of holding an instance reference.
2. TelemetryReporter is a trait, there are 3 concrete implementations: NoopTelemetryReporter, DriverTelemetryReporter and ExecutorTelemetryReporter. TelemetryReporter.getTelemetryReporter() will get an appropriate instance to send log message. (Details is mentioned in TelemetryReporter.scala)